### PR TITLE
Avoid clang warnings.

### PR DESCRIPTION
**Pull request by toregge**
@baldersheim : please review

### CODE REVIEW

File: eval/src/vespa/eval/tensor/mixed/packed_mappings_builder.cpp [hunk 1]
```diff
@@ -54,8 +54,8 @@ PackedMappingsBuilder::target_memory(char *mem_start, char *mem_end) const
     ssize_t avail_sz = mem_end - mem_start;
     assert(needs_sz <= avail_sz);
 
-    uint32_t * int_store_mem = (uint32_t *) mem_start;
-    uint32_t * offsets_mem = (uint32_t *) (mem_start + int_store_size);
+    uint32_t * int_store_mem = (uint32_t *) (void *) mem_start;
+    uint32_t * offsets_mem = (uint32_t *) (void *) (mem_start + int_store_size);
```
Comment (baldersheim) [inline eval/src/vespa/eval/tensor/mixed/packed_mappings_builder.cpp:58]: Is it not better that mem_start is void ?

File: eval/src/vespa/eval/tensor/mixed/packed_mixed_tensor_builder.cpp [hunk 2]
```diff
@@ -39,7 +39,7 @@ PackedMixedTensorBuilder<T>::build(std::unique_ptr<ValueBuilder<T>>)
 
     // copy cells:
     memcpy(cells_mem, &_cells[0], cells_size);
-    ConstArrayRef<T> cells((T *)cells_mem, _cells.size());
+    ConstArrayRef<T> cells((T *)(void *) cells_mem, _cells.size());
```
Comment (baldersheim) [inline eval/src/vespa/eval/tensor/mixed/packed_mixed_tensor_builder.cpp:42]: Is it not better that mem_start is void * ?

**Review by baldersheim — APPROVED** [review]

File: eval/src/vespa/eval/tensor/mixed/packed_mappings_builder.cpp (hunk 1 above)
Comment (toregge) [reply eval/src/vespa/eval/tensor/mixed/packed_mappings_builder.cpp:58]: If `mem_start` is `void*` then any code using `mem_start + offset` would need to use `((char *) mem_start) + offset`

File: storage/src/vespa/storage/storageserver/rpc/storage_api_rpc_service.cpp [hunk 3]
```diff
@@ -117,7 +117,8 @@ void compress_and_add_payload_to_rpc_params(mbus::BlobRef payload,
 
     params.AddInt8(comp_type);
     params.AddInt32(static_cast<uint32_t>(to_compress.size()));
-    params.AddData(buf.stealBuffer(), buf.getDataLen());
+    auto buffer_len = buf.getDataLen();
+    params.AddData(buf.stealBuffer(), buffer_len);
```
Comment (baldersheim) [inline storage/src/vespa/storage/storageserver/rpc/storage_api_rpc_service.cpp:121]: This is not the first time this treacherous interface gives us trouble.
Fixed in #14575

**Pull request merged**